src/main/java/tutoraid/MainApp.java

```diff
@@ -61,9 +61,11 @@ public void init() throws Exception {
 
         UserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(config.getUserPrefsFilePath());
         UserPrefs userPrefs = initPrefs(userPrefsStorage);
-        TutorAidStudentStorage tutorAidStudentStorage =
-                new JsonTutorAidStudentStorage(userPrefs.getStudentBookFilePath());
         TutorAidLessonStorage tutorAidLessonStorage = new JsonTutorAidLessonStorage(userPrefs.getLessonBookFilePath());
+        TutorAidStudentStorage tutorAidStudentStorage =
+                new JsonTutorAidStudentStorage(userPrefs.getStudentBookFilePath(),
+                        tutorAidLessonStorage.readLessonBook().orElseGet(SampleDataUtil::getSampleLessonBook));
```

Comment:
A bug now occurs when the user edits the lessons JSON file and edit one of the fields to be invalid. Because of this, `tutorAidLessonStorage.readLessonBook()` will throw `DataConversionException` and the app will crash. One proposed fix is
```suggestion
        TutorAidStudentStorage tutorAidStudentStorage;
        try {
            tutorAidLessonStorage.readLessonBook(userPrefs.getLessonBookFilePath());
            tutorAidStudentStorage =
                    new JsonTutorAidStudentStorage(userPrefs.getStudentBookFilePath(),
                            tutorAidLessonStorage.readLessonBook().orElseGet(SampleDataUtil::getSampleLessonBook));
        } catch (DataConversionException e) {
            tutorAidStudentStorage = new JsonTutorAidStudentStorage(userPrefs.getStudentBookFilePath(),
                    new LessonBook());
        }
```

However, the app will still crash since the line `Student student = initialStudent.toStudent(lessons)` in `JsonSerializableStudentBook` will throw an exception, since the lesson no longer exists and the lesson's students' lessonNames field still contains it. As such, refer to my other comment in `InitialStudent` for the continued proposed fix to solve this bug